### ISSUE TITLE
Ошибка анализа #Вставка и #Удаление в теле запроса для заимствование #ИзменениеИКонтроль

### ISSUE THREAD

**Issue by Kellendros007** [1C-Company/1c-edt-issues#565]
### Описание ошибки

Появление ошибки вида
Текст метода имеет отличия от базового метода, что не допустимо
   в случае расширения "Изменение и контроль"
при изменении (#Вставка/#Удаление) в тела заимствованного запроса используя механизм #ИзменениеИКонтроль

### Как воспроизвести

1. Добавить метод с запросом в коде в расширение выбрав режим Вставка с контролем
2. Добавить в тело запроса блок #Вставка #КонецВставки или #Удаление #КонецУдаления
3. Получить ошибку

### Ожидаемое поведение

Корректный анализ кода без вывода ошибки о отличии от исходной функции.

### Лог рабочей области

[trace.zip](https://github.com/1C-Company/1c-edt-issues/files/9144961/trace.zip)

### Версия 1С:EDT

Ruby 2021.3.4

### Операционная система

Windows

### Дополнительная информация

В конфигурации:
```
Функция Проверка()
	Запрос = Новый Запрос;
	Запрос.Текст =
	"ВЫБРАТЬ
	| Документ.Ссылка КАК Ссылка
	|ИЗ
	| Документ.ЗаказКлиента КАК Документ
	|";
	Возврат Запрос;
КонецПроцедуры
```

В расширении:
```
&ИзменениеИКонтроль("Проверка")
Функция Проверка()
	Запрос = Новый Запрос;
	Запрос.Текст =
	"ВЫБРАТЬ
	| Документ.Ссылка КАК Ссылка
	#Вставка
	| Документ.Номер КАК Номер
	#КонецВставки
	|ИЗ
	| Документ.ЗаказКлиента КАК Документ
	|";
	Возврат Запрос;
КонецПроцедуры
```

**Comment by Jimmi910** [comment]
Так текс запроса нельзя разрывать такими вставками.
Хочешь изменить запрос, скопируй его в отдельную процедуру, поправь как нужно, и после текста в основной процедуре подмени текст запроса

`#Вставка
Запрос.Текст = МойТекстЗапроса();
#КонецВстаки`

**Comment by NikitaMikhaylovSB** [comment]
Такой запрос может нормально отработать только в рантайме, пример проблемы:
```
	"ВЫБРАТЬ
	#Удалить
	| Документ.Ссылка КАК Ссылка
	#КонецУдаления
	#Вставка
	| Документ.Номер КАК Номер
	#КонецВставки
```

Вот такой запрос уже конструктор не сможет понять нормально. А если еще будет несколько таких изменений...
Вердикт: и правильно, что ругается!

**Comment by Kellendros007** [comment]
> Так текс запроса нельзя разрывать такими вставками. Хочешь изменить запрос, скопируй его в отдельную процедуру, поправь как нужно, и после текста в основной процедуре подмени текст запроса
> 
> `#Вставка Запрос.Текст = МойТекстЗапроса(); #КонецВстаки`

Запрет на разрыв запроса с использованием вставки следует из документации или ограничений платформы?



> Такой запрос может нормально отработать только в рантайме, пример проблемы:
> 
> ```
> 	"ВЫБРАТЬ
> 	#Удалить
> 	| Документ.Ссылка КАК Ссылка
> 	#КонецУдаления
> 	#Вставка
> 	| Документ.Номер КАК Номер
> 	#КонецВставки
> ```
> 
> Вот такой запрос уже конструктор не сможет понять нормально. А если еще будет несколько таких изменений... Вердикт: и правильно, что ругается!

Если запрос состоит из сотни строк, а мне надо в него внедрить 2 своих поля, зачем мне жертвовать читабельностью кода и заставлять разбираться в различиях между двумя запросами?

**Comment by NikitaMikhaylovSB** [comment]
> Если запрос состоит из сотни строк, а мне надо в него внедрить 2 своих поля, зачем мне жертвовать читабельностью кода и заставлять разбираться в различиях между двумя запросами?
Технически в рантайме отработает, не хотите сделать правильно в разработке - никто не запрещает, можно в исходниках нарисовать что угодно.

**Comment by marmyshev** [comment]
> Запрет на разрыв запроса с использованием вставки следует из документации или ограничений платформы?

Да. такое ограничение есть и всегда было в документации и синтаксис-помощнике платформы. Даже [специально стандарт](https://its.1c.ru/db/v8std#content:439:hdoc:_top:разрыв%20препроцессором) для особо не-читающих СП написали, чтоб уж совсем вопросов не было.

**Comment by Kellendros007** [comment]
> > Запрет на разрыв запроса с использованием вставки следует из документации или ограничений платформы?
> 
> Да. такое ограничение есть и всегда было в документации и синтаксис-помощнике платформы. Даже [специально стандарт](https://its.1c.ru/db/v8std#content:439:hdoc:_top:%D1%80%D0%B0%D0%B7%D1%80%D1%8B%D0%B2%20%D0%BF%D1%80%D0%B5%D0%BF%D1%80%D0%BE%D1%86%D0%B5%D1%81%D1%81%D0%BE%D1%80%D0%BE%D0%BC) для особо не-читающих СП написали, чтоб уж совсем вопросов не было.

Если вы считаете что предложенные по данной ссылке примеры (где идет разрыв кода вставками области и условий сильно усложняющий анализ логики кода) аналогичны тому примеру что я привел (где такая вставка может упростить анализ изменений кода не нарушая его читабельность) и при этом вы расцениваете такую механику как серьезную ошибку (не достойную уровня предупреждения с возможностью его скрытия), ваше право, настаивать не буду. Но как по мне, есть рекомендации, а есть ограничения, и если что то может дать, в одном случае пользу, а в другом вред, это должно быть предупреждением, но никак не ошибкой, если в категорию ошибки попадает и полезная реализация.

**Comment by agibalovsa** [comment]
https://github.com/1C-Company/1c-edt-issues/issues/645
